### DISCUSSION TITLE
How can I migrate project settings and presets & bookmarks?

### DISCUSSION THREAD

**Discussion by abdonrd** [directus/directus#11829] [Archive]
I can migrate the schema with the commands:

- `npx directus schema snapshot schema.yaml`
- `npx directus schema apply schema.yaml`

But there are any option to migrate the `Project Settings`? And the `Presets & Bookmarks`?

**Comment by azrikahar** [comment]
There isn't a CLI/direct way to migrate data yet, but currently you can get all of them via the API and add them in via the API again or via custom migration files (you might need to strip things like ID/primary key for example if you're adding to a new instance rather than updating an existing one).

**Reply by abdonrd** [reply]
Ok! Thanks Azri!

**Comment by EdouardDem** [comment]
@abdonrd @azrikahar 
### 

I've developed a CLI tool for synchronizing Directus settings, available here: https://github.com/tractr/directus-sync.

It's crafted to facilitate the export and import of Directus configurations like schema, `dashboards`, `flows`, `folders`, `operations`, `panels`, `permissions`, `presets`, `roles`, `settings`, `translations`, and `webhooks`, organizing them into JSON files for repository storage.

This method ensures a safe, incremental update process. For a detailed walkthrough, please consult the tutorial I authored: https://medium.com/tractr/streamlining-directus-workflows-with-directus-sync-a-comprehensive-guide-baf33b9cc399.

I hope it helps.
